qiskit/quantum_info/synthesis/one_qubit_decompose.py

```diff
@@ -138,18 +148,31 @@ def __init__(self, basis="U3", use_dag=False):
     def build_circuit(self, gates, global_phase):
         """Return the circuit or dag object from a list of gates."""
         qr = [Qubit()]
+        lookup_gate = False
+        if len(gates) > 0 and isinstance(gates[0], tuple):
+            lookup_gate = True
+
```

Comment:
Are there any situations in our internal calls that would lead to `lookup_gate` being false?  I guess because it's a public method we technically need to leave this `lookup_gate=False` path, although logically I think it's actually a private method so we could deprecate it and move it to be private.

Reply:
Yeah, I left it in place specifically because it was a public method and I was worried someone downstream was doing something weird and using this as a utility method with the old format. This is only needed if we're assuming someone could have been calling this from outside the class.

Reply:
I think that if we don’t think that method’s used we should probably deprecate it so we don’t have to keep supporting an implementation detail we don’t use any more, but we can do that in a follow-up.